packages/alfa-style/src/property/background.ts

```diff
@@ -70,45 +126,712 @@ export namespace Background {
+    ),
+    Iterable.flatten
+  );
+
+  const parseBackgroundLayerList = map(
+    filter(
+      separatedList(
+        parseBackgroundLayer,
+        delimited(option(Token.parseWhitespace), Token.parseComma)
+      ),
+      (layers) =>
+        [...layers]
+          .slice(0, -1)
+          .every((layer) =>
+            [...layer].every(([property]) => property !== "background-color")
+          ),
+      () => "Only the last layer may contain a color"
+    ),
+    (layers) => List.of(layers, ", ")
+  );
+
   /**
-   * @see https://drafts.csswg.org/css-backgrounds/#background
+   * @see https://drafts.csswg.org/css-backgrounds/#propdef-background
    */
   export const Shorthand = Property.Shorthand.of(
-    ["background-image", "background-color"],
-    map(Color.parse, (color) => {
-      return Record.of({
-        "background-color": color,
-        "background-image": Image.initial as Image.Specified,
-      });
+    [
+      "background-color",
+      "background-image",
+      "background-position-x",
+      "background-position-y",
+      "background-size",
+      "background-repeat-x",
+      "background-repeat-y",
+      "background-attachment",
+      "background-origin",
+      "background-clip",
+    ],
+    map(parseBackgroundLayerList, (layers) => {
+      let color: Option<Color.Specified> = None;
+      let image: Array<Image.Specified.Item> = [];
+      let positionX: Array<Position.X.Specified.Item> = [];
+      let positionY: Array<Position.Y.Specified.Item> = [];
+      let size: Array<Size.Specified.Item> = [];
+      let repeatX: Array<Repeat.X.Specified.Item> = [];
+      let repeatY: Array<Repeat.Y.Specified.Item> = [];
+      let attachment: Array<Attachment.Specified.Item> = [];
+      let origin: Array<Origin.Specified.Item> = [];
+      let clip: Array<Clip.Specified.Item> = [];
+
+      for (const layer of layers) {
+        for (const property of layer) {
+          switch (property[0]) {
+            case "background-color":
+              color = Option.of(property[1]);
+              break;
+            case "background-image":
+              image.push(property[1]);
+              break;
+            case "background-position-x":
+              positionX.push(property[1]);
+              break;
+            case "background-position-y":
+              positionY.push(property[1]);
+              break;
+            case "background-size":
+              size.push(property[1]);
+              break;
+            case "background-repeat-x":
+              repeatX.push(property[1]);
+              break;
+            case "background-repeat-y":
+              repeatY.push(property[1]);
+              break;
+            case "background-attachment":
+              attachment.push(property[1]);
+              break;
+            case "background-origin":
+              origin.push(property[1]);
+              break;
+            case "background-clip":
+              clip.push(property[1]);
+          }
```

Comment:
This also feels very redundant and could maybe use
```javascript
for (const [key, value] of layer) {
  result[key].push(value)
}
```
(also need to replace `color` by an array and take its last value).

Reply:
That's not really possible for the same reason as https://github.com/Siteimprove/alfa/pull/301/files/ac71fefe5f47951cec8c139a6a31ecef8546ddc8#r468449936. In Servo, this stuff is generated via Python templating to avoid having to spell it out while still appeasing the type checker 🙈